secret/secret.go

```diff
@@ -0,0 +1,96 @@
+package secret
+
+import (
+	"log"
+	"os"
+
+	"github.com/SUSE/scf-secret-generator/model"
+	"github.com/SUSE/scf-secret-generator/password"
+	"github.com/SUSE/scf-secret-generator/ssh"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	corev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/rest"
+)
+
+// The name of the secret stored in the kube API
+const SECRET_NAME = "secret"
+
+func GetSecrets() corev1.SecretInterface {
+	// Set up access to the kube API
+	kubeConfig, err := rest.InClusterConfig()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	clientSet, err := kubernetes.NewForConfig(kubeConfig)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	return clientSet.CoreV1().Secrets(os.Getenv("KUBERNETES_NAMESPACE"))
+}
+
+func UpdateSecrets(s corev1.SecretInterface, secrets *v1.Secret, create, dirty bool) {
+	if create {
+		_, err := s.Create(secrets)
+		if err != nil {
+			log.Fatal(err)
+		}
+		log.Println("Created `secret`")
+	} else if dirty {
+		_, err := s.Update(secrets)
+		if err != nil {
+			log.Fatal(err)
+		}
+		log.Println("Updated `secret`")
+	}
+}
+
+func GetOrCreateSecret(s corev1.SecretInterface) (create bool, secrets *v1.Secret) {
```

Comment:
Should probably be called `GetOrCreateSecrets` to match `GenerateSecrets` and `UpdateSecrets`.

Reply:
Fixed in a6a5a76